app/api/services/brief_responses.py

```diff
@@ -10,15 +10,19 @@ def __init__(self, *args, **kwargs):
         super(BriefResponsesService, self).__init__(*args, **kwargs)
 
     def get_brief_responses(self, brief_id, supplier_code):
-        responses = (db.session.query(BriefResponse.created_at,
-                                      BriefResponse.id,
-                                      BriefResponse.brief_id,
-                                      BriefResponse.supplier_code,
-                                      Supplier.name.label('supplier_name'))
-                     .join(Supplier)
-                     .filter(BriefResponse.supplier_code == supplier_code,
-                             BriefResponse.brief_id == brief_id,
-                             BriefResponse.withdrawn_at.is_(None))
-                     .all())
+        query = (
+            db.session.query(BriefResponse.created_at,
+                             BriefResponse.id,
+                             BriefResponse.brief_id,
+                             BriefResponse.supplier_code,
+                             Supplier.name.label('supplier_name'))
+            .join(Supplier)
+            .filter(
+                BriefResponse.brief_id == brief_id,
+                BriefResponse.withdrawn_at.is_(None)
+            )
+        )
+        if supplier_code:
+            query.filter(BriefResponse.supplier_code == supplier_code)
```

Comment:
i have a feeling filter is immutable

Reply:
Good catch! I think you're right - when I debug the query it doesn't seem to change when a `supplier_code` value is passed in.